distributed/tests/test_worker.py

```diff
@@ -1634,3 +1634,22 @@ def bad_heartbeat_worker(*args, **kwargs):
                 else:
                     assert w.status == "closed"
     assert "Heartbeat to scheduler failed" in logger.getvalue()
+
+
+@gen_cluster(client=True, nthreads=[("127.0.0.1", 1)])
+async def test_delete_data(c, s, w):
+    df = dask.datasets.timeseries(dtypes={"x": int, "y": float}, freq="1s").reset_index(
+        drop=True
+    )
+
+    df2 = df.persist()
+    await df2
+
+    del df
+    assert len(w.data) > 0
```

Comment:
Note that this deletion almost certainly hasn't gone through yet.  I'm not sure what you're trying to test here.

Reply:
Actually, can you help explain the objective of this test?  Are there not already tests that verify that data gets deleted from workers?

Reply:
What I'm trying to test here is that `Worker.delete_data` is awaited (see https://github.com/dask/distributed/issues/3920#issue-643386582), or sync in the case of this PR. The test here is a simplification of the test in https://github.com/rapidsai/dask-cuda/blob/5c7942d6a62d630475fa51dd6b80fcf469bca142/dask_cuda/tests/test_spill.py#L358-L411, but I also noticed that the test here does not reproduce that error 100% accurately. I'm thus not sure that error will be easy to reproduce, so I'm open to suggestions on a better way to handle this here, should we remove this test and just make `delete_data` sync, as is the focus of this PR?

Reply:
Friendly ping here @mrocklin . :)

Reply:
Hi @pentschev .  Thanks for the friendly ping.  My apologies for the delay.

What I'm hearing is that you want to test that Dask deletes data once references to that data are deleted.  I would be very surprised if we didn't already have tests for this.  It might be worth looking around in the test_scheduler.py or test_client.py file if that is your goal.

After looking at the test that you pointed to briefly I'm afraid to say that I wasn't able to understand the intent of that test.  I'm not sure how to help on that topic.

Reply:
As mentioned, making `delete_data` sync seems to be the right thing to do. Before that there were corner cases where it would not be awaited and current tests in distributed seem not to pick this case up, I was trying to come up with a small test to ensure that, but it has proven very challenging, even though the dask-cuda test did pick that up. I'm fine with just removing the test completely but keeping the change (removing `async` from `delete_data`). What do you say about this?

Reply:
> What do you say about this?

I have high confidence that we're already testing data deletion.  I'm entirely ok dropping this test entirely.

Reply:
Done.